### PR TITLE
wayland: don't depend on the order of global announcements

### PR DESCRIPTION
**Pull request by mahkoh**
E.g. wl_subcompositor could be announced before wl_compositor.

### CODE REVIEW

**Comment by Dudemanguy** [comment]
It might not be technically mandated by the spec but I'm not sure there is ever a scenario where subcompositor could happen before compositor? Same thing with viewporter happening before compositor.

**Comment by mahkoh** [comment]
This happens when globals are stored in a hashmap which makes the order random.

File: video/out/wayland_common.c [hunk 1]
```diff
@@ -1832,6 +1828,14 @@ int vo_wayland_init(struct vo *vo)
...
         return false;
     }
+    if (wl->subcompositor) {
+        wl->video_subsurface = wl_subcompositor_get_subsurface(wl->subcompositor, wl->video_surface, wl->surface);
+        wl_subsurface_set_desync (wl->video_subsurface);
+    }
+    if (wl->viewporter) {
+        wl->viewport = wp_viewporter_get_viewport (wl->viewporter, wl->surface);
+        wl->video_viewport = wp_viewporter_get_viewport (wl->viewporter, wl->video_surface);
+    }
```
Comment (Dudemanguy) [inline video/out/wayland_common.c:1838]: Could you move these so they occur after the `if (create_xdg_surface(wl))` part? Also no spaces between the parenthesis and the function.

File: video/out/wayland_common.c [hunk 2]
```diff
@@ -1843,6 +1839,15 @@ int vo_wayland_init(struct vo *vo)
     if (create_xdg_surface(wl))
         return false;
 
+    if (wl->subcompositor) {
+        wl->video_subsurface = wl_subcompositor_get_subsurface(wl->subcompositor, wl->video_surface, wl->surface);
+        wl_subsurface_set_desync(wl->video_subsurface);
+    }
+    if (wl->viewporter) {
```
Comment (Dudemanguy) [inline video/out/wayland_common.c:1846]: One last thing. Could you put a newline inbetween the two if statements?

**Comment by Dudemanguy** [comment]
Thanks!

**Pull request merged**